tests/cli-rustup.rs

```diff
@@ -2065,3 +2065,53 @@ warning: If you meant to build software to target that platform, perhaps try `ru
         );
     });
 }
+
+/// Checks that `rust-toolchain.toml` files are considered
+#[test]
+fn rust_toolchain_toml() {
+    setup(&|config| {
+        let cwd = config.current_dir();
+        let toolchain_file = cwd.join("rust-toolchain.toml");
+        raw::write_file(&toolchain_file, "[toolchain]\nchannel = \"nightly\"").unwrap();
+
+        expect_stdout_ok(config, &["rustc", "--version"], "hash-nightly-2");
```

Comment:
I'm not certain but I think `nightly` is the default in this kind of test.  Probably worth ensuring that the output of `rustc --version` is not `hash-nightly-2` before you write the toolchain file, and is that afterwards.  To demonstrate that it's the toolchain file which has caused it.